src/algebra/field.lean

```diff
@@ -5,46 +5,95 @@ Authors: Robert Lewis, Leonardo de Moura, Johannes Hölzl, Mario Carneiro
 -/
 import algebra.ring.basic
 import algebra.group_with_zero
+
+/-!
+# Division (semi)rings, (semi)fields
+
+In this file we define four typeclasses:
+
+* `division_semiring` is a nontrivial `semiring` with multiplicative inverses for nonzero elements;
+* `division_ring` is a nontrivial `ring` with multiplicative inverses for nonzero elements,
+  e.g., quaternions;
+* `semifield` is a commutative `division_semiring`, e.g., non-negative real numbers `ℝ≥0`;
+* `field` is a commutative `division_ring`, e.g., `ℚ`, `ℝ`, `ℂ`, `ℤ_[p]` etc.
+
+## Tags
+
+division ring, field
+-/
 open set
 
 set_option old_structure_cmd true
 
 universe u
 variables {K : Type u}
 
-/-- A `division_ring` is a `ring` with multiplicative inverses for nonzero elements -/
-@[protect_proj, ancestor ring has_inv]
-class division_ring (K : Type u) extends ring K, div_inv_monoid K, nontrivial K :=
+/-- A `division_semiring` is a `semiring` with multiplicative inverses for nonzero elements. -/
+@[protect_proj, ancestor semiring div_iv_monoid nontrivial]
+class division_semiring (K : Type u) extends semiring K, div_inv_monoid K, nontrivial K :=
 (mul_inv_cancel : ∀ {a : K}, a ≠ 0 → a * a⁻¹ = 1)
 (inv_zero : (0 : K)⁻¹ = 0)
 
-section division_ring
-variables [division_ring K] {a b : K}
+section division_semiring
 
-/-- Every division ring is a `group_with_zero`. -/
+variables [division_semiring K] {a b : K}
+
+/-- Every division semiring is a `group_with_zero`. -/
 @[priority 100] -- see Note [lower instance priority]
-instance division_ring.to_group_with_zero :
+instance division_semiring.to_group_with_zero :
   group_with_zero K :=
-{ .. ‹division_ring K›,
-  .. (infer_instance : semiring K) }
+{ .. ‹division_semiring K› }
```

Comment:
I know it's out of scope here, but... Isn't this just `division_semiring.to_semiring.to_group_with_zero`?